capstone/src/main/java/com/google/step/clients/RestaurantClient.java

```diff
@@ -0,0 +1,113 @@
+package com.google.step.clients;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.step.data.Restaurant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class RestaurantClient {
+  private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /**
+   * Puts a single entity of any kind into datastore.
+   * @param restaurant, to be processed into an Entity and put into datastore
+   * @param email, separate from Restaurant class and passed from UserService in the servlet
+   */
+  public void putRestaurant(Restaurant restaurant, String email) {
+    // Retrieve restaurant properties
+    long id = restaurant.getRestaurantKey();
+    String name = restaurant.getName();
+    GeoPt location = restaurant.getLocation();
+    String story = restaurant.getStory();
+    List<String> cuisineList = restaurant.getCuisine();
+    String phone = restaurant.getPhone();
+    String website = restaurant.getWebsite();
+    String status = restaurant.getStatus();
+
+    // Parse into a RestaurantInfo Entity
+    Entity restaurantInfo = new Entity("RestaurantInfo", id);
+    restaurantInfo.setProperty("restaurantKey", id);
+    restaurantInfo.setProperty("name", name);
+    restaurantInfo.setProperty("email", email);
+    restaurantInfo.setProperty("location", location);
+    restaurantInfo.setProperty("story", story);
+    restaurantInfo.setProperty("cuisine", cuisineList);
+    restaurantInfo.setProperty("phone", phone);
+    restaurantInfo.setProperty("website", website);
+    restaurantInfo.setProperty("status", status);
+
+    // The following value is hardcoded while we implement the properties.
+    restaurantInfo.setProperty("score", 2.5);
+
+    datastore.put(restaurantInfo);
+  }
+
+  /**
+   * Queries for a single restaurant in datastore based on restaurantKey.
+   * @param restaurantKey, unique key of the restaurant to query for
+   * @return Restaurant, parsed from the datastore properties
+   * @throws NoSuchElementException if no RestaurantInfo entity is found with the requested key
+   */
+  public Restaurant getSingleRestaurant(long restaurantKey) {
+    Query query = new Query("RestaurantInfo")
+                      .setFilter(new Query.FilterPredicate(
+                          "restaurantKey", Query.FilterOperator.EQUAL, restaurantKey));
+    PreparedQuery results = datastore.prepare(query);
+    Entity resultEntity = results.asSingleEntity();
```

Comment:
I believe this causes a problem because the restaurantKey is the same as the usersId, since when created we just grab the userId. Because of this when calling the query we will probably get all the restaurants by the user instead, so the asSingleEntity() method won't really work. Another workaround is we can change how we store the restaurantId to just be something random and that solves the problem as well.

Reply:
Corrected!